src/main/frontend/components/StarRating.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from "react"
+import Star from "./Star"
+
+const StarRating = ({ formRating, handleRatingChange }) => {
+  const [rating, setRating] = useState(parseInt(formRating) || 0)
+  const [selection, setSelection] = useState(0)
+
+  const hoverOver = event => {
+    let val = 0
+    if (event && event.target && event.target.getAttribute("data-star-id"))
+      val = event.target.getAttribute("data-star-id")
+    setSelection(val)
+  }
+
+  const handleClick = event => {
+    setRating(event.target.getAttribute("data-star-id") || rating)
+  }
+  
+  useEffect(() => {
+    handleRatingChange(rating.toString())
+  }, [rating])
+
+  return (
+    <div
+      onMouseOut={() => hoverOver(null)}
+      onClick={handleClick}
+      onMouseOver={hoverOver}
+      value={formRating}
+      name="rating"
+      id="rating"
+    >
+      <label htmlFor="rating">Rating: 
+      {Array.from({ length: 5 }, (v, i) => (
+        <Star
+          starId={i + 1}
+          key={`star_${i + 1}`}
+          marked={selection ? selection >= i + 1 : rating >= i + 1}
+          value={i + 1}
+        />
+      ))}
```

Comment:
Let's move this array of `Stars` to above the return statement on 23 and set it to a variable like `starComponents`.  Then we can display `{starComponents}` to keep this more readable